src/main/java/org/magicdgs/hdf5j/io/address/FileAddressManager.java

```diff
@@ -0,0 +1,177 @@
+        final byte[] undefinedBytes = new byte[addressSize];
+        for (int i = 0; i < addressSize; i++) {
+            undefinedBytes[i] = -1;
+        }
+        this.undefinedAddress = new FileAddress(undefinedBytes);
+        this.addressSize = addressSize;
+    }
+
+    /**
+     * Gets the size of the addresses decoded/encoded by this file address manager.
+     *
+     * @return size of addresses/offsets.
+     */
+    public int getAddressSize() {
+        return addressSize;
+    }
+
+    /**
+     * Gets the undefined address for this FileManager.
+     *
+     * @return cached undefined file address.
+     */
+    public FileAddress getUndefinedAddress() {
+        return undefinedAddress;
+    }
+
+    /**
+     * Seeks the byte chanel to the provided file address.
+     *
+     * @param byteChannel channel to seek.
+     * @param address     file address to set the pointer to.
+     *
+     * @throws HDF5jException.FileAddressException if the address cannot be handled or it is
+     *                                             undefined.
+     * @throws IOException                         if an IO error occurs.
+     */
+    public void seek(final SeekableByteChannel byteChannel, final FileAddress address)
+            throws IOException {
+        if (undefinedAddress.equals(address)) {
+            throw new HDF5jException.FileAddressException(address, " cannot be used to seek");
+        }
+        byteChannel.position(validateAddress(address).position);
+    }
+
+    /**
+     * Encodes the provided address into the {@link ByteBuffer}.
+     *
+     * @param address address to encode.
+     * @param buffer  buffer to put the address bytes.
+     *
+     * @return the same buffer passed to the method.
+     *
+     * @throws HDF5jException.FileAddressException if there is a problem encoding the address.
+     */
+    public ByteBuffer encodeAddress(final FileAddress address, final ByteBuffer buffer) {
+        // validate the address to be used
+        final FileAddress validated = validateAddress(address);
+
+        Preconditions.checkArgument(buffer != null, "null buffer");
+        // for encoding, we require to be able to write at least addressSize bytes
+        Preconditions.checkArgument(buffer.remaining() >= addressSize,
+                "required at least %s bytes in the buffer to write an address", addressSize);
+
+        if (validated.equals(undefinedAddress)) {
+            // if it is undefined, it is independent of the size of the arrays
+            buffer.put(undefinedAddress.bytes);
+        } else if (validated.bytes.length == addressSize) {
+            buffer.put(validated.bytes);
+        } else if (validated.bytes.length < addressSize) {
+            // pad the byte array with zeroes at the beginning to get the same position encoded with more bytes
+            IntStream.range(validated.bytes.length, addressSize).forEach(i -> buffer.put((byte) 0));
```

Comment:
This code will be required for the method to get from a long, so maybe we can generalize...